apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/nodes/NodesWebscript1.java

```diff
@@ -587,21 +587,42 @@ public void createNode(final CreateNodeOptions createNodeOptions, WebScriptRespo
                     @Override
                     public Object execute() throws Throwable {
                         NodeRef parent = new NodeRef(createNodeOptions.parent);
-                        NodeRef copyFrom = null;
+                        MetadataChanges metadataChanges;
 
-                        if (createNodeOptions.copyFrom != null) {
-                            copyFrom = new NodeRef(createNodeOptions.copyFrom);
+                        if (!nodeService.exists(parent)) {
+                            response.setStatus(404);
+                            writeJsonResponse(response, "Parent does not exist");
+                            return null;
                         }
 
                         NodeRef nodeRef;
-                        if (copyFrom == null) {
+                        NodeRef copyFrom = null;
+                        if (createNodeOptions.copyFrom == null) {
                             nodeRef = nodeService
-                                    .createNode(parent, createNodeOptions.properties,
-                                            new QName(createNodeOptions.type),
-                                            null);
+                                    .createNode(parent, createNodeOptions.name,
+                                            new QName(createNodeOptions.type));
                         } else {
+                            copyFrom = new NodeRef(createNodeOptions.copyFrom);
+                            if (!nodeService.exists(copyFrom)) {
+                                response.setStatus(404);
+                                writeJsonResponse(response, "CopyFrom does not exist");
+                                return null;
+                            }
                             nodeRef = nodeService.copyNode(copyFrom, parent, true);
                         }
+
+                        if (createNodeOptions.type != null) {
+                            metadataChanges = new MetadataChanges(new QName(createNodeOptions.type),
```

Comment:
I think the difference between this line and line 618 is only the type of the node. So you can just keep a variable for the type and use that when calling the constructor of the MetadataChanges.

Something like this:
                        QName type;
                        if (createNodeOptions.type == null) {
                            type = new QName(createNodeOptions.type);
                        }
                        else if (createNodeOptions.type == null && createNodeOptions.copyFrom != null) {
                            type = nodeService.getMetadata(copyFrom).type;
                        }
                        else {
                            response.setStatus(400);
                            writeJsonResponse(response, "Please provide parameter: type when creating a new node");
                            return null;
                        }
                        MetadataChanges metadataChanges = new MetadataChanges(type,
                                null, null, createNodeOptions.properties);